dbms/programs/server/TCPHandler.cpp

```diff
@@ -173,8 +191,11 @@ void TCPHandler::runImpl()
             else
                 processOrdinaryQuery();
 
-            sendEndOfStream();
+            /// Reset BlockIO in advance to log destruction actions
+            state.io.reset();
```

Comment:
There is a problem with this invocation.
If an exception is thrown blow this line, then we will become here https://github.com/yandex/ClickHouse/pull/2482/files#diff-8389b5ff47b68bdd1959bf673e944fb4R203 and call onException() which will use destroyed object state.io.

Reply:
Thanks! First I will simply remove it.